account-trigger-zip-code.cls

```diff
@@ -17,3 +18,67 @@ trigger populateSchoolDistrictBasedOnZip on Account (before update, before inser
                       }
     }
 }
+
+     /* 
+
+Trigger should be after update/insert
+
+1. Loop through to add zip codes to a set. Filtering by using Record Type = church, zip is filled
+    and parent account is blank. Declared the Set outside of for loop due to scope.
+
+2. Run a soql to find all matched Parent Accounts with same zip codes 
+
+3. Create a map of Account with zip code as key
+
+4. Populate the Account lookup with the key value.
+
+    Set<String> zipCodeSet = new Set<String>();
+
+    for (Account a : [SELECT BillingPostalCode
+                        FROM Account
+                        WHERE RecordTypeId = '0121N000000qpORQAY'
+                        AND Id IN :Trigger.new
+                        AND ParentId = null //ParentId or DistrictLookupId?
+                        AND BillingPostalCode != null]) {
+
+                            zipCodeSet.add(a.BillingPostalCode);
+                        }
+    
+    List<Account> matchedSchoolDistricts = new List<Account>([SELECT Id, BillingPostalCode
```

Comment:
Create a List(matchedSchoolDistricts) of Accounts whose zip code matches that of the ones in zipCodeSet. Filters added to check if record type is School District.